CMakeLists.txt

```
@@ -485,8 +485,9 @@ if(CONFIG_USERSPACE)
 endif()
 
 get_property(TOPT GLOBAL PROPERTY TOPT)
-set_ifndef(  TOPT -Wl,-T) # clang doesn't pick -T for some reason and complains,
-                          # while -Wl,-T works for both, gcc and clang
+get_property(COMPILER_TOPT TARGET compiler PROPERTY linker_script)
```

combined with other comments, this should be on linker target.
```suggestion
get_property(COMPILER_TOPT TARGET linker PROPERTY linker_script)
```

---

yeah, was pretty unsure about this -- there's a difference between the compiler driver (gcc/clang/cc) and the linker (bfd, ld, lld). This particular flag gets applied to the compiler driver, not directly to the linker?